src/compiler/scanner.rs

```diff
@@ -1,5 +1,44 @@
+#![allow(non_upper_case_globals)]
+
+use std::array::IntoIter;
+use std::collections::HashMap;
+use std::iter::FromIterator;
+
 use crate::{CharacterCodes, SyntaxKind};
 
+lazy_static! {
+    static ref text_to_keyword_obj: HashMap<&'static str, SyntaxKind> =
+        HashMap::from_iter(IntoIter::new([
+            ("false", SyntaxKind::FalseKeyword),
+            ("true", SyntaxKind::TrueKeyword),
+        ]));
+}
+
+lazy_static! {
+    static ref text_to_keyword: HashMap<&'static str, SyntaxKind> = text_to_keyword_obj.clone();
+}
+
+fn is_unicode_identifier_start(ch: char) -> bool {
+    false
+}
+
+fn is_identifier_start(ch: char) -> bool {
+    ch >= CharacterCodes::A && ch <= CharacterCodes::Z
+        || ch >= CharacterCodes::a && ch <= CharacterCodes::z
+        || ch == CharacterCodes::dollar_sign
```

Comment:
In something a bit meta, in the Typescript version this is called `CharacterCodes::$`/`CharacterCodes::_` because those are allowed as "identifier starts" in TS/JS, whereas in Rust I guess they're not (or solitary `_` is reserved/special?) so had to spell them out as `dollar_sign`/`underscore`

Reply:
Ah yeah, `_` is the "ignore this" variable/param name in Rust, so that makes sense you can't use it.